WooCommerce/Classes/POS/Presentation/ItemListView.swift

```diff
@@ -35,7 +39,18 @@ struct ItemListView: View {
             })
         }
         .refreshable {
+            // Only allows one refresh task at a time now that the refresh control is released on redraw.
+            guard refreshControlTaskID == nil else {
+                return
+            }
+            refreshControlTaskID = .init()
+        }
+        .task(id: refreshControlTaskID) {
+            guard refreshControlTaskID != nil else {
+                return
+            }
             await posModel.loadItems(base: .root)
+            refreshControlTaskID = nil
```

Comment:
Would such an approach work? Or does it end up having similar problems as a detached task?
```suggestion
            await Task {
                 await posModel.loadItems(base: .root)
             }.value
```

Reply:
This does work, at least from a quick test. Perhaps it's a better workaround for now since it leaves the UI as we'd expect it?

There is some suggestion on Stack Overflow that it could cause hard to diagnose issues, because the refresh won't get cancelled when the view's lifetime ends. I tried to break it, and couldn't, and I think the risk of that with a refresh control is quite low.

Reply:
Thanks for suggesting this @staskus! I tested it and it worked on my end, it's a better workaround because the refresh control is shown while awaiting the async task and the changes are simpler. The lack of cancellation when the view lifetime ends is a downside, but given the impact this tradeoff seems small.

Reply:
Updated in https://github.com/woocommerce/woocommerce-ios/pull/14870/commits/0e52b0f810da499be07e74622f8ea8839cbe7de3.